catalog/spatial/registry/registry-report-viewer/src/main/java/org.codice.ddf.registry.report.viewer/RegistryReportViewer.java

```diff
@@ -0,0 +1,486 @@
+                                internationalStringTypeHelper.getString(extrinsicObjectType.getName());
+                    }
+
+                    collectionMap.put(collectionName, collectionValues);
+                }
+            } else {
+                Map<String, Object> collectionValues = new HashMap<>();
+
+                collectionValues.putAll(getCustomSlots(extrinsicObjectType.getSlot()));
+
+                String collectionName = "";
+                if (extrinsicObjectType.isSetName()) {
+                    collectionName =
+                            internationalStringTypeHelper.getString(extrinsicObjectType.getName());
+                }
+
+                collectionMap.put(collectionName, collectionValues);
+            }
+        }
+
+        Map<String, Object> extrinsicInfoMap = new HashMap<>();
+        extrinsicInfoMap.put(GENERAL, generalMap);
+        extrinsicInfoMap.put(COLLECTIONS, collectionMap);
+
+        return extrinsicInfoMap;
+    }
+
+    private Map<String, Object> getOrganizationInfo(RegistryPackageType registryPackage) {
+        Map<String, Object> organizationMap = new HashMap<>();
+
+        List<OrganizationType> organizationTypes = registryPackageTypeHelper.getOrganizations(
+                registryPackage);
+        for (OrganizationType organizationType : organizationTypes) {
+            Map<String, Object> organizationInfo = new HashMap<>();
+            Map<String, Object> contactInfo = new HashMap<>();
+
+            String orgName = "";
+
+            if (organizationType.isSetName()) {
+                orgName = internationalStringTypeHelper.getString(organizationType.getName());
+            }
+
+            addNonEmptyKeyValue(contactInfo,
+                    ADDRESSES,
+                    getAddresses(organizationType.getAddress()));
+            addNonEmptyKeyValue(contactInfo,
+                    PHONE_NUMBERS,
+                    getPhoneNumbers(organizationType.getTelephoneNumber()));
+            addNonEmptyKeyValue(contactInfo,
+                    EMAIL_ADDRESSES,
+                    getEmailAddresses(organizationType.getEmailAddress()));
+
+            organizationInfo.put(CONTACT_INFO, contactInfo);
+            organizationInfo.put(CUSTOM_SLOTS, getCustomSlots(organizationType.getSlot()));
+            organizationMap.put(orgName, organizationInfo);
+        }
+
+        return organizationMap;
+    }
+
+    private Map<String, Object> getContactInfo(RegistryPackageType registryPackage) {
+        Map<String, Object> contactMap = new HashMap<>();
+        List<PersonType> persons = registryPackageTypeHelper.getPersons(registryPackage);
+        for (PersonType person : persons) {
+            Map<String, Object> personMap = new HashMap<>();
+            Map<String, Object> contactInfo = new HashMap<>();
+
+            PersonNameType personName = person.getPersonName();
+            ArrayList<String> nameArray = new ArrayList<>();
```

Comment:
Same basic thought as above. While it's cheap to create an ArrayList, if you're going to break out of this loop with the next test, it's cheaper still to skip its creation. Move below the test.